### ISSUE TITLE
Multiple accounts directories multiplies the number of detected replays

### ISSUE THREAD

**Issue by thomas-ashcraft** [eivindveg/HotSUploader#96]
I have 5 HOTS account directories on system right now. The total number of replays of about 2,000. When I started HotSUploader i saw this:

![hotslogs uploader fx bug mult accounts](https://cloud.githubusercontent.com/assets/6583664/11898446/e53bc3cc-a5a8-11e5-8ce1-4bcd429861eb.png)

After removing the extra directories (and shortcuts to them), I get the correct calculation of the replay files.

**Comment by zhedar** [comment]
Sounds like the smurfs are upon us :)

What do you mean by removing extra directories? Which exactly did you remove? Did you move all of your replays in just one folder?
I currently cannot reproduce this, because I only use one account, but maybe @eivindveg can have a look at it.

**Comment by eivindveg** [comment]
I've smurfed too and noticed it. Forgot to open an issue. The problem is that the account is known locally, but the account id gets exchanged with Battletag name with HotSLogs.com. For some reason, we add it as an account with empty string afterwards. This should be prevented. That's one side of the issue.

The multiplication I've noticed as well; I have no theories about that yet. Currently swamped with exams, so I'll look into that too; this really shouldn't be the case.

**Comment by foxi** [comment]
Quick note: This happens with a single account as well.
I accidentally logged in to another region and the game created another (empty) Hero directory inside the account directory. Since then, the uploader shows every new replay twice.

**Comment by zhedar** [comment]
Nice to know that, so I can actually reproduce this issue without creating a smurf.  
May look into that then.

**Comment by zhedar** [comment]
Ok, I found the issue and got a quick fix the problem, but it I need to investigate the real flaw, which causes this. Atleast we now know, where it happens and can probably fix that in an elegant way.

**Comment by barrett777** [comment]
Some more info on this that may be related:

If a user logs into another region, their BattleTag is (usually) the same for both regions - so BattleTags aren't unique overall, just per region

**Comment by foxi** [comment]
Indeed, the directory structure basically looks like:
`<Documents Folder>\Heroes of the Storm\`

This contains 2 shortcut links to the accoount folder:
```
T_<num11>_<num12>@1 => Accounts\<AccountId>\1-Hero-1-<num13>
T_<num21>_<num22>@2 => Accounts\<AccountId>\2-Hero-1-<num23>

```

The "1-Hero..." folder contains only empty "Replays\Multiplayer\" directories (I guess 1 is the NA game region, 2 is Europe, where I'm located), while the other has the data for my real account.

**Comment by zhedar** [comment]
Yeah, may client created a 3-Hero-1-*** folder for Asia as well and the duplication isn't due to confusion of battletags, it happens at startup, when new replay files are created and a database check duplicates them (or even worse with more accounts).

**Comment by zhedar** [comment]
This happened, because for every account directory we (well, to be honest... I introduced this myself in #93 to fix #91) pulled all saved `ReplayFiles` from the db, thus duplicating the hell out of them.

I replaced this by an integrity check, where DB entries are created or removed for every account directory and pulling the whole refreshed data *once* after that. Should made the code clearer and a bit faster just by the way :)
@eivindveg Maybe you could review and test that, it worked quite well for me.

**Comment by eivindveg** [comment]
Fixed fixed in 463f974. Cherry-picking to hotfix/2.0.2
